mailpoet/lib/Automation/Integrations/WooCommerce/Fields/OrderFieldsFactory.php

```diff
@@ -386,10 +386,20 @@ private function getProductOptions(): array {
       ARRAY_A
     );
 
-    return array_map(function ($product) {
+    /** @var \WC_Product_Data_Store_Interface $dataStore */
+    $dataStore = (new \WC_Product())->get_data_store();
```

Comment:
@websupporter Is there a reason not to modify the products query above rather than fetching product type for each product (I suppose 1 query per product)?